neetbox/utils/environment.py

```diff
@@ -113,7 +119,60 @@ def watcher_fun(env_instance: Environment, do_update_gpus: bool):
                 target=watcher_fun, args=(self, self._with_gpu), daemon=True
             )
             self._watcher.start()
-
+            
+    def _run(self, command):
+        """Returns (return-code, stdout, stderr)"""
+        p = subprocess.Popen(command, stdout=subprocess.PIPE,
+                            stderr=subprocess.PIPE, shell=True)
+        raw_output, raw_err = p.communicate()
+        rc = p.returncode
+        if self.platform_info["architecture"] == "32bit":
+            enc = 'oem'
+        else:
+            enc = locale.getpreferredencoding()
+        output = raw_output.decode(enc)
+        err = raw_err.decode(enc)
+        
+        return rc, output.strip(), err.strip()
+    
+    def _get_nvidia_smi(self):
+        return "nvidia-smi"
+    
+    def get_gpu_device(self):
+        smi = self._get_nvidia_smi()
+        rc, out, _ = self._run(smi + " -L")
+        gpu_list = out.split("\n")
+        
+        return gpu_list
+    
+    def get_cuda_version(self):
+        conda_list = self.get_conda_packages()
+        cuda = [p.split(",")[2] for p in conda_list if p.startswith("cudatoolkit")]
+        if len(cuda) == 1:
+            return "CUDA runtime version: " + cuda[0].strip()
+        
+        nvcc = "nvcc --version"
+        rc, out, _ = self._run(nvcc)
+        match = re.search(r'release .+ V(.*)', out)
+        if match is None:
+            return None
+        return "CUDA runtime version: " + match.group(1).strip()
+    
+    def get_conda_packages(self):
```

Comment:
conda packages are placed in conda's virtual env dir, it's better if we use `pkg.is_installed` to determine if a package is installed. Adding conda integrations is welcomed, but do not add any 'not-pure-python' things here in `neetbox.utils`. We should add these codes in `neetbox.integrations`. Anyway, please remove the code about conda and we will talk about it later.